apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/template-selector-item.js

```diff
@@ -26,18 +26,27 @@ const TemplateSelectorItem = props => {
 		blocks = [],
 	} = props;
 
-	// Define static or dynamic preview.
-	const innerPreview = useDynamicPreview ? (
-		<Disabled>
-			<BlockPreview blocks={ blocks } viewportWidth={ 960 } />
-		</Disabled>
-	) : (
-		<img
-			className="template-selector-item__media"
-			src={ staticPreviewImg }
-			alt={ staticPreviewImgAlt }
-		/>
-	);
+	const renderInnerButton = () => {
+		if ( ! blocks.length ) {
+			return 'blank' !== value ? <Spinner /> : null;
```

Comment:
Just wondering whether this component should be aware of the concept of `blank`?

Reply:
yes, it's a good point. Let's find a way to remove that checking.